packages/manager/apps/key-management-service/src/pages/credential/generalInformations/delete/DeleteCredentialModal.page.spec.tsx

```diff
@@ -0,0 +1,146 @@
+    }),
+    useRouteSynchro: vi.fn(),
+    ShellContext: React.createContext({
+      environment: {
+        getEnvironment: vi.fn(() => ({
+          getUser: vi.fn().mockReturnValue({ ovhSubsidiary: 'FR' }),
+          getUserLocale: vi.fn().mockReturnValue('fr_FR'),
+        })),
+      },
+    }),
+  };
+});
+
+const deleteModalTitleLabel =
+  labels.credentials.key_management_service_credential_delete_modal_headline;
+
+describe('Credential delete modal test suite', () => {
+  const mockNavigate = vi.fn();
+
+  beforeEach(() => {
+    mockNavigate.mockReset();
+    vi.spyOn(router, 'useNavigate').mockImplementation(() => mockNavigate);
+  });
+
+  test('should display the delete modal', async () => {
+    const { container } = await renderTestApp(mockPageUrl);
+
+    // Check modal is opened
+    await waitFor(() => {
+      assertOdsModalVisibility({ container, isVisible: true });
+      expect(screen.getByText(deleteModalTitleLabel)).toBeVisible();
+    }, WAIT_FOR_DEFAULT_OPTIONS);
+  });
+
+  test('should navigate and show a notification after successful deletion', async () => {
+    const { container } = await renderTestApp(mockPageUrl);
+
+    // Wait for modal to open
+    await waitFor(() => {
+      expect(screen.getByText(deleteModalTitleLabel)).toBeVisible();
+    }, WAIT_FOR_DEFAULT_OPTIONS);
+
+    await changeOdsInputValue({
+      inputLabel:
+        labels.credentials
+          .key_management_service_credential_delete_modal_input_label,
+      inputValue: 'TERMINATE',
+    });
+
+    const submitButton = await getOdsButtonByLabel({
+      container,
+      label: 'Résilier', // Label from MRC
+      timeout: 3000,
+      disabled: false,
+    });
+
+    await waitFor(() => userEvent.click(submitButton));
+
+    // Check navigation
+    await waitFor(() => {
+      expect(mockNavigate).toHaveBeenCalledWith(mockCredentialListPageUrl, {
+        state: { deletingCredentialId: mockCredentialItem.id },
+      });
+    });
+
+    // Check notification
+    await waitFor(() => {
+      expect(
+        screen.getByText(
+          labels.credentials.key_management_service_credential_delete_success,
+        ),
+      ).toBeVisible();
+    });
+  });
+
+  test('should navigate and show a notification after failed deletion', async () => {
+    const { container } = await renderTestApp(mockPageUrl, {
+      isCredentialDeleteKO: true,
+    });
+
+    // Wait for modal to open
+    await waitFor(() => {
+      expect(screen.getByText(deleteModalTitleLabel)).toBeVisible();
+    }, WAIT_FOR_DEFAULT_OPTIONS);
+
+    await changeOdsInputValue({
+      inputLabel:
+        labels.credentials
+          .key_management_service_credential_delete_modal_input_label,
+      inputValue: 'TERMINATE',
+    });
+
+    const submitButton = await getOdsButtonByLabel({
+      container,
+      label: 'Résilier',
+      timeout: 3000,
+      disabled: false,
+    });
+
+    await waitFor(() => userEvent.click(submitButton));
```

Comment:
We don't waitFor a click.

You can waitFor the button is visible and enabled. But waitFor can click multiple times.

https://testing-library.com/docs/user-event/v13/#api

And you can setup userEvent it's recommended (But I known it's rarelly done : https://testing-library.com/docs/user-event/intro#writing-tests-with-userevent)